File: eRecipes/eRecipes.Model/Recept.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eRecipes.Model
{
    public class Recept
    {
        public int ReceptId { get; set; }

        public string Naziv { get; set; }

        public string? OpisRecepta { get; set; }

        public byte[]? Slika { get; set; }
        public int VrstaJelaId { get; set; }

        public int KategorijaId { get; set; }
        public int? VrijemePripreme { get; set; }
        public  VrstaJela VrstaJela{get;set;}

        public  Kategorija Kategorija { get; set; }
        public int? KorisnikId { get; set; }
        public Korisnik Korisnik { get; set; }

        public DateTime? DatumObjave { get; set; }

        public bool? Premium { get; set; }
        public bool? Status { get; set; }
        public string? StateMachine { get; set; }
        public ICollection<ReceptSastojak> ReceptSastojaks { get; set; } = new List<ReceptSastojak>();

    }
}

File: eRecipes/UI/erecipes_desktop/lib/models/recept.dart
import 'package:erecipes_desktop/models/kategorija.dart';
import 'package:erecipes_desktop/models/korisnik.dart';
import 'package:erecipes_desktop/models/recept_sastojak.dart';
import 'package:erecipes_desktop/models/vrsta_jela.dart';
import 'package:json_annotation/json_annotation.dart';

part 'recept.g.dart';

@JsonSerializable()
class Recept{
  int? receptId;
  String? naziv; 
  String? slika;
  int? vrstaJelaId;
  int? kategorijaId;
  int? vrijemePripreme;
  int? korisnikId;
  String? opisRecepta;
  DateTime? datumObjave;
  Korisnik? korisnik;
  Kategorija? kategorija;
  VrstaJela? vrstaJela;
  List<ReceptSastojak>? sastojci; 
  Recept({this.receptId, this.naziv});

  factory Recept.FromJson(Map<String,dynamic> json)=> _$ReceptFromJson(json);

 Map<String,dynamic> toJson() => _$ReceptToJson(this);
}